src/main/java/duke/DukeReader.java

```diff
@@ -0,0 +1,78 @@
+package duke;
+
+import task.Deadline;
+import task.Event;
+import task.Task;
+import task.ToDo;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.util.ArrayList;
+import java.util.Scanner;
+import java.util.regex.Pattern;
+
+public class DukeReader {
+    private static ArrayList<Task> tasks = new ArrayList<>();
+
+    public static void readFromFile() {
+        try {
+            File f = new File("duke.txt");
+            Scanner scanner = new Scanner(f);
+            while (scanner.hasNext()) {
+                createTask(scanner.nextLine());
+            }
+        } catch (FileNotFoundException e) {
+            System.out.println("File not found");
+            e.printStackTrace();
+        }
+
+    }
+
+    private static void createTask(String input) {
+        String[] parts = input.split(Pattern.quote(" | "));
+        String type = parts[0];
+        boolean isDone = Integer.parseInt(parts[1]) == 1;
+        String description = parts[2];
+        switch (type) {
+        case "T":
+            createToDo(isDone, description);
+            break;
+        case "D":
+            String by = parts[3];
+            createDeadline(isDone, description, by);
+            break;
+        case "E":
+            String at = parts[3];
+            createEvent(isDone, description, at);
+            break;
```

Comment:
A small indentation error here. According to our coding standards, there is no indentation for case clauses. See [Coding Standards](https://se-education.org/guides/conventions/java/basic.html#conditionals)